plugins/inputs/t128_graphql/response_processor.go

```diff
@@ -13,56 +13,287 @@ type ProcessedResponse struct {
+		processedRoot = append(processedRoot, leaf)
+	}
+
+	return processedRoot
+}
+
+func processChildren(jsonData *gabs.Container, mode string, path string, fields map[string]string, tags map[string]string) ([]*ProcessedResponse, error) {
+	output := []*ProcessedResponse{}
+
+	processChild := func(child *gabs.Container, path string) {
+		processedChild := processNode(child, path, fields, tags)
+		for _, mergedChildOutput := range mergeAll(processedChild) {
+			output = append(output, mergedChildOutput)
+		}
+	}
+
+	processAsMap := func() ([]*ProcessedResponse, error) {
+		children, err := jsonData.ChildrenMap()
+		if err == nil {
+			for key, child := range children {
+				processChild(child, path+"/"+key)
 			}
+			return output, nil
+		}
+		return nil, fmt.Errorf("could not process map")
+	}
 
-			if isNil(data) {
-				return nil, fmt.Errorf("found empty data for collector %s: field %s", collectorName, fieldName)
+	processAsList := func() ([]*ProcessedResponse, error) {
+		children, err := jsonData.Children()
+		if err == nil {
+			for _, child := range children {
+				processChild(child, path)
 			}
-			processedResponse.Fields[fieldRenamed] = data
+			return output, nil
 		}
+		return nil, fmt.Errorf("could not process list")
+	}
 
-		for tagRenamed, tagName := range tags {
-			data := node[tagName]
+	if mode == "map" {
+		return processAsMap()
+	}
+	return processAsList()
+}
 
-			if strings.Contains(tagName, "/") {
-				nestedObj, err := child.JSONPointer("/" + tagName)
-				if err != nil {
-					return nil, fmt.Errorf("unexpected response for collector %s: tag %s", collectorName, tagName)
-				}
-				data = nestedObj.Data()
-			}
+/*
+Definitions:
+leaf - a single tag/field stored in *ProcessedResponse
+branch - any *ProcessedResponse that isn't a leaf
+*/
+func collectLeaf(leaf interface{}, mode string, path string, lookup map[string]string) (*ProcessedResponse, error) {
+	output := newResponse()
 
-			if isNil(data) {
-				processedResponse.Tags[tagRenamed] = ""
-				continue
+	if leafName, ok := lookup[path]; ok {
+		if !isNil(leaf) {
+			if mode == "field" {
+				output.Fields[leafName] = leaf
+			} else {
+				output.Tags[leafName] = fmt.Sprintf("%v", leaf)
 			}
+			return output, nil
+		}
+	}
+	return nil, fmt.Errorf("could not collect leaf")
+}
+
+// merges all leaves/branches at a given node
+func mergeAll(itemsToMerge []*ProcessedResponse) []*ProcessedResponse {
+	leaves := []*ProcessedResponse{}
+	branches := []*ProcessedResponse{}
 
-			processedResponse.Tags[tagRenamed] = fmt.Sprintf("%v", data)
+	//fill branches and leaves structs
+	for _, response := range itemsToMerge {
+		if getResponseSize(response) > 1 {
+			branches = append(branches, response)
+		} else {
+			leaves = append(leaves, response)
 		}
+	}
 
-		processedResponses = append(processedResponses, processedResponse)
+	//happens if all itemsToMerge are leaves
+	if len(branches) == 0 {
+		return mergeLeaves(leaves)
 	}
 
-	return processedResponses, nil
+	//happens
```

Comment:
happens? i dont tihnk this really helps the User follow the code unless the comment is more descriptive

Reply:
Should be `happens if itemsToMerge are branches and leaves`